gatsby-node.js

```diff
@@ -12,6 +14,78 @@ const paginationPath = (page, totalPages, prefix = `/`) => {
   }
 }
 
+const getAlbumWithTitleId = curry((title, id) => {
+  return {
+    path: mangoSlugfy(title),
+    component: path.resolve(`./src/templates/album.js`),
+    context: {
+      albumId: id,
+      slugPath: mangoSlugfy(title),
+    },
+  }
+});
+
+
+const getPaginatedPageWithContentComponentPathPrefix = curry((content, component, pathPrefix, postsPerPage, tagKeyName, prefix) => {
+  const posts = content;
+  if (!isEmpty(posts)) {
+    const numPages = Math.ceil(posts.length / postsPerPage);
+    return Array.from({ length: numPages }).map((_, i) => ({
+        path: paginationPath(i, numPages, pathPrefix),
+        component: component,
+        context: {
+          limit: postsPerPage,
+          skip: i * postsPerPage,
+          numPages,
+          currentPage: i + 1,
+          prevPath: paginationPath(i - 1, numPages, pathPrefix),
+          nextPath: paginationPath(i + 1, numPages, pathPrefix),
+          tag: !isEmpty(tagKeyName) ? tagKeyName : null,
+          prefix: !isEmpty(prefix) ? prefix : ``,
+        },
+      }));
+  }
+  return [];
+});
+
+const getPaginatedPageWithContentComponentEmptyPrefix = getPaginatedPageWithContentComponentPathPrefix(__, __, ``, 3, null, null);
+const getPaginatedPageWithContentComponentGridPrefix = getPaginatedPageWithContentComponentPathPrefix(__, __, `/grid/`, 9, null, null);
+const getPaginatedTagWithContentComponentTagPrefix = (keyname) => getPaginatedPageWithContentComponentPathPrefix(__, __, `/tag/${keyname}/`, 3, keyname, `tag/${keyname}/`);
+const getPaginatedTagWithContentComponentTagGridPrefix = (keyname) => getPaginatedPageWithContentComponentPathPrefix(__, __, `/tag/${keyname}/grid/`, 9, keyname, `tag/${keyname}/`);
+
+const getKeyname = prop('keyname');
+const getPaginatedList = fn => fn(__, path.resolve("./src/templates/list.js"));
```

Comment:
Some arrow functions with one parameter has parenthesis and some don't, try to keep consistent.